### PR TITLE
DynamoDB: Add software test for decoding list of objects

### PR DESCRIPTION
**Pull request by amotl**
## About
People reported problems converging a list of objects from DynamoDB to CrateDB. Adding a test case to evaluate the situation.

## References
- https://github.com/crate/commons-codec/pull/37

### CODE REVIEW

File: tests/io/dynamodb/test_copy.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                    "date": {"S": "2024-08-28T20:05:42.603Z"},
+                    "utm_adgroup": {"L": [{"S": ""}, {"S": ""}]},
+                    "utm_campaign": {"S": "34374686341"},
+                    "utm_medium": {"S": "foobar"},
+                    "utm_source": {"S": "google"},
+                }
+            }
+        ]
+    },
+}
```
Comment (amotl) [inline tests/io/dynamodb/test_copy.py:18]: Based on this input data (DynamoDB table record), and a SQL DDL like
```sql
CREATE TABLE IF NOT EXISTS foobar (data OBJECT(DYNAMIC));
```

[...]

File: tests/io/dynamodb/test_copy.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        "utmTags": [
+            {
+                "date": "2024-08-28T20:05:42.603Z",
+                "utm_adgroup": ["", ""],
+                "utm_campaign": "34374686341",
+                "utm_medium": "foobar",
+                "utm_source": "google",
+            }
+        ],
+    }
```
Comment (amotl) [inline tests/io/dynamodb/test_copy.py:54]: .... that's the outcome that proves data converges into CrateDB well, afaics.

**Review by wierdvanderhaar — APPROVED** [review]
Let get this out.

**Comment by amotl** [comment]
Thanks for acknowledging. It's just a software test. No code has been adjusted, so this will not yield or need another release.

**Pull request merged**